newton/_src/solvers/kamino/_src/core/builder.py

```
@@ -485,6 +500,8 @@ def add_joint_descriptor(self, joint: JointDescriptor, world_index: int = 0) ->
         # Check if the descriptor is valid
         if not isinstance(joint, JointDescriptor):
             raise TypeError(f"Invalid joint descriptor type: {type(joint)}. Must be `JointDescriptor`.")
+        if joint.dof_type == JointDoFType.FREE and any(value > 0.0 for value in joint.f_j):
+            msg.warning("Ignoring joint friction on FREE joint %r.", joint.name)
```

Ruff (0.16.1)

[error] Undefined name `msg`

(F821)